### PR TITLE
[wip] add mate to kube stack

### PR DESCRIPTION
**Pull request by linki**
**work in progress**

this adds mate by default to the stack 

discussion needed:
* opensource mate?
* hosted zone needs to be configured correctly
* `--no-sync` is the default but explicitly defined for clarity
* `route:53*` policy stuff must obviously be tweaked

### CODE REVIEW

File: cluster/userdata-master.yaml [hunk 1]
```diff
@@ -756,6 +760,36 @@ write_files:
...
+          spec:
+            containers:
+            - name: mate
+              image: pierone.stups.zalan.do/teapot/mate:v0.0.4-31-gebb7747
+              args:
+              - --producer=kubernetes
+              - --kubernetes-domain=mate.teapot.zalan.do.
+              - --consumer=aws
+              - --aws-hosted-zone=mate.teapot.zalan.do.
+              - --no-sync
```
Comment (linki) [inline cluster/userdata-master.yaml]: mate knows about producers (source of records, e.g. kube services) and consumers (dns providers, e.g. route53)

it seems the zone is duplicated but that's not the full story. `--kubernetes-domain` is responsible for what dns records to create. `--aws-hosted-zone` just picks a zone on the aws side. (legacy from google)

**Comment by szuecs** [comment]
:+1:

File: cluster/userdata-master.yaml [hunk 2]
```diff
@@ -756,6 +760,36 @@ write_files:
...
+            - name: mate
+              image: pierone.stups.zalan.do/teapot/mate:v0.0.4-32-g0af5e78
+              args:
+              - --producer=kubernetes
+              - --kubernetes-domain=HOSTED_ZONE.
+              - --consumer=aws
+              - --aws-hosted-zone=HOSTED_ZONE.
+              - --no-sync
+            - name: kubectl
+              image: quay.io/coreos/hyperkube:v1.3.6_coreos.0
```
Comment (hjacobs) [inline cluster/userdata-master.yaml]: we should upgrade hyperkube version here

File: cluster/userdata-master.yaml [hunk 3]
```diff
@@ -756,6 +760,36 @@ write_files:
...
+      spec:
+        replicas: 1
+        template:
+          metadata:
+            labels:
+              app: mate
+          spec:
+            containers:
+            - name: mate
+              image: pierone.stups.zalan.do/teapot/mate:v0.0.4-32-g0af5e78
```
Comment (hjacobs) [inline cluster/userdata-master.yaml]: I think mate should be in the OSS registry (and we should open source) --- having Pier One here depends on correct credentials

File: cluster/userdata-master.yaml (hunk 2 above)
Comment (Raffo) [inline cluster/userdata-master.yaml]: 1.4.3 is the latest on quay, 1.4.4 not yet there.

File: cluster/userdata-master.yaml (hunk 3 above)
Comment (szuecs) [inline cluster/userdata-master.yaml]: Then we should also publish mate

Comment (Raffo) [inline cluster/userdata-master.yaml]: We're on the same page: https://github.bus.zalan.do/teapot/mate/issues/23

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**